libraries/tools/kotlin-stdlib-gen/src/templates/Aggregates.kt

```diff
@@ -346,6 +346,7 @@ object Aggregates : TemplateGroupBase() {
             if (!iterator.hasNext()) return null
 
             var minElem = iterator.next()
+            if (!iterator.hasNext()) return minElem
             var minValue = selector(minElem)
             while (iterator.hasNext()) {
```

Comment:
Condition check now can be moved to the end of the loop by replacing `while(...) { }` loop with `do { } while(...)` to avoid checking `iterator.hasNext()` twice on the first iteration.